### DISCUSSION TITLE
Specify personal objective function

### DISCUSSION THREAD

**Discussion by FlorianJ0** [nlmixrdevelopment/nlmixr#559] [General]
Dear nlmixr developer team,
I am working on a pk problem with n compartments where the input is a time varying blood concentration. The thing is, my objective function is made of 2 parts : sum of n compartment concentration VS measurement #1 and concentration in one of the compartment vs measurement #2.
I cannot find how to do this in nlmixr, if it is even possible ?
Many thanks,
Florian

**Comment by mattfidler** [comment]
Hi Florian,

It is possible, but not yet through the simple function that is used in most of `nlmixr`.  This is discussed here:

https://github.com/nlmixrdevelopment/nlmixr/issues/261
https://github.com/nlmixrdevelopment/nlmixr/issues/355

We are working toward having likelihood integrated into nlmixr functions,  but it is not ready yet

**Reply by FlorianJ0** [reply]
Hi, thank you for the answer and the links. So I read those threads and started to play around the solutions. As far I understand, I should use the gnlmm syntax in which the llik block is accessible. I am not familiar with those syntaxes, but is this a correct approach ? (totally random model and parameters setup)
```R
library(nlmixr)
ode <- "
foo = CL;
d/dt(depot) =-KA*depot;
d/dt(centr) = KA*depot - KE*centr;
d/dt(out) = KE*centr - KO*out;
"
sys1 <- RxODE(ode)

pars <- function() {
  WT <- WT[1]
  CL <- exp(THETA[1]+0.0000000000001*log(WT/65) + ETA[1]) 
  KA <- exp(THETA[2] + ETA[2])
  KE <- exp(THETA[3])
  KO <- exp(THETA[4])
  V <- CL / KE
  sig2 <- exp(THETA[5])
}

llik <- function() {
  pred <- centr + depot + out
  dnorm(DV, pred, sd = sqrt(sig2), log = TRUE) #1st term of the objective fcn
  dnorm(DV_out, out , sd = sqrt(sig2), log = TRUE) #2nd term of the objective fcn
}
inits <- list(THTA = c(-3.22, 0.47, -2.45,1, 0))
inits$OMGA <- list(ETA[1]+ETA[2]~c(.027, .01, .37))


fit <- gnlmm(llik, theoPerso, inits, pars, sys1, control = list(trace = TRUE, nAQD = 1))
```
Thanks, 
Florian

**Reply by mattfidler** [reply]
I think you are trying to do the following:

```r
llik <- function() {
  pred <- centr + depot + out
  dnorm(DV, pred, sd = sqrt(sig2), log = TRUE) + dnorm(DV_out, out , sd = sqrt(sig2), log = TRUE) #2nd term of the objective fcn
}
```

I think this is only currently supported by `gnlmm` not `gnlmm2`.  It also presupposes a specific data structure.

**Reply by FlorianJ0** [reply]
Thank you very much ! I am working on this.
Best,
Florian
